Including fetch

thestrauss3 opened the pull request.

thestrauss3 committed:
Fix error messages for new burger form

error messages display correctly
form retains data from submission
rails routes can handle resubmission

thestrauss3 committed:
FactoryGirl and specs

Add factory girl and implement in all test suites
fix tests to increase coverall of code
fix new restaurant form to display errors and retain information

thestrauss3 committed:
Merge branch 'master' of https://github.com/thestrauss3/five-guys-project into factory_girl

thestrauss3 committed:
added react

thestrauss3 committed:
react

thestrauss3 committed:
successful react add-on

thestrauss3 committed:
testing refactor

thestrauss3 committed:
cleaning up spaces

thestrauss3 committed:
working on api

thestrauss3 committed:
Refactored some code

Refactored, fixed factories to not have faker

thestrauss3 committed:
Stop doing stupid controller stuff

thestrauss3 committed:
Fix restaurant, hours and dining type allow nil

thestrauss3 committed:
working on seeds.rb

nwalberts commented on app/controllers/api/fortunes_controller.rb:
```diff
@@ -0,0 +1,35 @@
+      "Another good night not to sleep in a eucalyptus tree."
+    ]
+  end
+end
```
You should not have this in your app?

nwalberts commented on app/controllers/restaurants_controller.rb:
```diff
@@ -9,7 +9,7 @@ def show
 
   def new
-    @restaurant = Restaurant.new
+    @restaurant = Restaurant.new()
```
Please fix this!

nwalberts commented on app/controllers/reviews_controller.rb:
```diff
@@ -6,7 +6,7 @@ def index
   def new
     @burger = Burger.find(params[:burger_id])
-    @review = Review.new
+    @review = Review.new()
```
please fix this!

nwalberts commented on app/views/burgers/show.html.erb:
```diff
@@ -6,7 +6,10 @@
+
+
 <% unless @burger.reviews.empty? %>
+
```
Please remove extra lines

nwalberts commented on app/views/static_pages/index.html.erb:
```diff
@@ -1,3 +1,10 @@
+
+
+<div id="something">
+</div>
```
What is this div doing?

nwalberts commented on config/routes.rb:
```diff
@@ -1,13 +1,16 @@
+  post 'burgers' => 'burgers/new'
+  post 'reviews' => 'reviews/new'
+  post 'restaurants' => 'restaurants/new'
 
```
what are these routes doing?

nwalberts commented on db/seeds.rb:
```diff
@@ -8,9 +8,16 @@
+  3.times do
+    FactoryGirl.create(:review, burger: b)
+  end
+end
```
Please rename "b" and "r" to more telling names

thestrauss3 committed:
hang on

nwalberts commented on react/src/containers/App.js:
```diff
@@ -0,0 +1,37 @@
+          <Route path="/reviews" component={ReviewIndex} />
+          <Route path='/restaurants' component={RestaurantIndex} />
+          <Route path='/restaurants/:id' component={RestaurantShow} />
+          <Route path='burgers/new' component={BurgerNew} />
```
inconsistent routes (last route), please fix. 

You also are getting a bit ahead of yourself. This is a LARGE PR. In the future. E.g. make a PR with ONLY the index page, then ONLY the show page, etc

thestrauss3 committed:
fixed seeds

thestrauss3 committed:
Fixed fetch burgers index

display burger names on burger index page through fetch as well as rails
removed unused and unnecessary files

enthusiastick commented on app/controllers/api/v1/burgers_controller.rb:
```diff
@@ -0,0 +1,10 @@
+  def show
+    @burger = Burger.all.sample
+    render json: @burger
+  end
```
So this action ignores the `:id` that is passed in through params, in favor of returning a random burger? Interesting.

enthusiastick commented on app/controllers/api_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class ApiController < ApplicationController
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :null_session
```
As a best practice, I prefer `protect_from_forgery unless: -> { request.format.json? }`. But this works too.

enthusiastick commented on spec/features/burgers_features/burger_add_spec.rb:
```diff
@@ -1,17 +1,34 @@
-    fill_in 'Price', with: "four dollars"
-    fill_in 'Description', with: "A seasame seed bun, swiss cheese, and two beef patties slathered in Garlic Butter"
+    fill_in 'Price', with: 4
+    fill_in 'Description', with: Faker::Internet.ip_v6_address
```
This seems like a strange choice.

enthusiastick commented:
This is a big refactor, touching several different parts of the code base. In the future, I would rather see PRs that focus on one feature at a time.
